plugins/sources/local.js

```diff
@@ -42,10 +44,15 @@ Local.prototype._read = function(){
 
     // if there is an error store it on the image object and pass it along
     if (err) {
-      // FIXME Add some kind of load-attempts counter to prevent infinite loop
-      _this.filePath = env.LOCAL_FILE_PATH + '/' + env.IMAGE_404;
-      _this._read();
-      return;
+      // if this is our second loop, accept defeat and error out
+      if (_this.readAttemptCount > 1) {
```

Comment:
Why not call it `attemptErrorsCount` (or something nicer) since it's only used in this one place? Then the increment function could be moved to line 47 and live a bit more close to its usage.